File: api/src/main/java/org/ftt/familytasktracking/entities/Reward.java
package org.ftt.familytasktracking.entities;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.GenericGenerator;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;
import java.util.UUID;

/**
 * Reward
 */
@Entity
@Table(name = "reward")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Reward {

    @Id
    @GeneratedValue(generator = "uuid2")
    @GenericGenerator(name = "uuid2", strategy = "org.hibernate.id.UUIDGenerator")
    private UUID uuid;

    private Integer cost;

    private String name;

    private String description;

    private Boolean redeemed;

    @CreationTimestamp
    private LocalDateTime createdAt;

    @UpdateTimestamp
    private LocalDateTime updatedAt;

    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    private Household household;

    @PrePersist
    @PreUpdate
    private void prePersist() {
        if (redeemed == null) {
            redeemed = false;
        }
    }

}

File: frontend/types/reward.ts
export interface Reward {
  uuid?: string;
  cost?: number;
  name?: string;
  description?: string;
  createdAt?: string;
  updatedAt?: string;
  redeemedAt?: string;
  /**
   * This is backend wise a bit confusing
   * redeemedBy is an uuid of the profile who
   * redeemed the reward.
   */
  redeemedBy?: string;
  redeemed?: boolean;
}

export interface RewardRequest {
  name?: string;
  description?: string;
  cost?: number;
  redeemed?: boolean;
}
